cassandane/Cassandane/Instance.pm

```diff
@@ -3242,16 +3250,19 @@ sub new_jmaptester_for_user ($self, $user, $new_arg = undef) {
     my $port = $arg{port};
     my $scheme = $arg{scheme};
 
-    require Cassandane::JMAPTester;
+    eval "require $tester_class; 1" || Carp::croak("can't load $tester_class: $@");
 
     # This causes all requests and responses to be printed to STDERR.
     local $ENV{JMAP_TESTER_LOGGER} = 'HTTP:-2'
         unless exists $ENV{JMAP_TESTER_LOGGER};
 
-    my $jtest = Cassandane::JMAPTester->new({
+    my $jtest = $tester_class->new({
         fallback_account_id => $user->username,
         default_using => [ @DEFAULT_USING ],
         %overrides,
+        ( $tester_class eq 'Cassandane::JMAPTesterWS'
+            ? (cache_connection => 1, ws_api_uri => '') : ()
+        ),
```

Comment:
This is really up to you, but I don't like that we're checking the $tester_class, here.  I think better would be to move the setting of cache_connection and ws_api_uri into `new_jmaptester_ws_for_user` so that the generic method is truly generic, and the specialization happens in the specialization code.